tests/rptest/tests/partition_movement_test.py

```diff
@@ -164,65 +156,37 @@ def test_static(self):
 
         self.logger.info(f"Creating topics...")
         for spec in topics:
-            self.client().create_topic(spec)
+            self._rpk_client.create_topic(spec.name, spec.partition_count,
+                                          spec.replication_factor)
 
         num_records = 1000
-        produced = set(
-            ((f"key-{i:08d}", f"record-{i:08d}") for i in range(num_records)))
 
+        rw_workload = RWWorkload(self.test_context, self.redpanda)
+        rw_workload.start()
         for spec in topics:
-            self.logger.info(f"Producing to {spec}")
-            producer = KafProducer(self.test_context, self.redpanda, spec.name,
-                                   num_records)
-            producer.start()
-            self.logger.info(
-                f"Finished producing to {spec}, waiting for producer...")
-            producer.wait()
-            producer.free()
-            self.logger.info(f"Producer stop complete.")
+            self.logger.info(f"Start producing to {spec}")
+            rw_workload.remote_start(spec.name, self.redpanda.brokers(),
+                                     spec.name, spec.partition_count)
+        for spec in topics:
+            self.logger.info(f"waiting for 100 r&w operations to warm up")
+            rw_workload.ensure_progress(spec.name, 100, 10)
 
+        self.logger.info(f"initiate movement")
         for _ in range(25):
             self._move_and_verify()
```

Comment:
It looks like after I switched to continuous workload (`rw_workload`) the test gets stuck and doesn't leave joint consensus withing 1m30s. 

```
[INFO  - 2022-07-22 04:04:21,788 - partition_movement - _do_move_and_verify - lineno:119]: replacement for topic4-0:1: [{'node_id': 2, 'core': 0}] -> [{'node_id': 1, 'core': 1}]
[INFO  - 2022-07-22 04:04:21,788 - partition_movement - _do_move_and_verify - lineno:122]: new assignments for topic4-0: [{'node_id': 1, 'core': 1}]
...
[INFO  - 2022-07-22 04:04:22,478 - partition_movement - derived_done - lineno:87]: derived assignments for topic4-0: [{'node_id': 2, 'core': 0}]
...
[INFO  - 2022-07-22 04:04:23,685 - partition_movement - derived_done - lineno:87]: derived assignments for topic4-0: [{'node_id': 1, 'core': 1}, {'node_id': 2, 'core': 0}]
...
[INFO  - 2022-07-22 04:05:52,194 - partition_movement - derived_done - lineno:87]: derived assignments for topic4-0: [{'node_id': 1, 'core': 1}, {'node_id': 2, 'core': 0}]
```

@dotnwat do you have an idea what to do with this test?

https://buildkite.com/redpanda/redpanda/builds/12891#018223e1-b1ec-4fbb-b6d8-f43255c15672

Reply:
asked @mmaslankaprv offline and he's fine with dropping it